app/src/App.js

```diff
@@ -1,144 +1,145 @@
 import React from 'react'
-import { useAragonApi, useConnectedAccount } from '@aragon/api-react'
+import { useAragonApi } from '@aragon/api-react'
 import {
-  Main,
   Button,
   IconPlus,
   Header,
   Split,
   GU,
   textStyle,
+  useLayout,
   useTheme,
   Tag,
   SyncIndicator,
 } from '@aragon/ui'
-import styled from 'styled-components'
-import { useAppLogic } from './app-logic'
+import { AppLogicProvider, useAppLogic } from './app-logic'
 import NoWrappedTokens from './screens/NoWrappedTokens'
 import Holders from './screens/Holders'
 import Panel from './components/ActionsPanel'
 import InfoBox from './components/InfoBox'
 import { IdentityProvider } from './components/IdentityManager/IdentityManager'
 
 function App() {
-  const { api, appState } = useAragonApi()
-  const {
-    wrappedTokenSymbol,
-    erc20TokenSymbol,
-    holders,
-    isSyncing,
-    orgTokenAddress,
-    wrappedTokenAddress,
-  } = appState
+  const { appState } = useAragonApi()
+  const { holders, isSyncing, outsideToken, wrappedToken } = appState
   const { actions, wrapTokensPanel, unwrapTokensPanel } = useAppLogic()
+  const { layoutName } = useLayout()
   const theme = useTheme()
-  let totalSuply = 0
-  if (holders && holders.length > 0) {
-    totalSuply = holders
-      .map(({ account, amount }) => parseFloat(amount))
-      .reduce((a, b) => a + b, 0)
-  }
```

Comment:
We generally prefer doing an on-chain query to get information like this, if it's available on-chain.

In the future, this can either become computationally expensive, or difficult if the `holders` array is paginated (because it grows too long).